### PR TITLE
Adding multiQC to amplicon-pipeline, dev -> main PR

### PR DESCRIPTION
**Pull request by chrisAta**
This PR pushes the changes from https://github.com/EBI-Metagenomics/amplicon-pipeline/pull/13 into main

### CODE REVIEW

File: conf/local.config [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*
+ * EMBL-EBI - Microbiome Informatics team - config file
```
Comment (mberacochea) [inline conf/local.config:2]: I would also add max cpus and memory, to cap proceses when running on a laptop:
```
 params {
   max_cpus   = 4
   max_memory = '8.GB'
   max_time   = '6.h'
}
```

File: conf/local.config [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+executor {
+    name = "local"
+}
+
+conda.enabled = false
+
+docker {
+    enabled = true
+    autoMounts = true
+    registry = "quay.io"
```
Comment (mberacochea) [inline conf/local.config:13]: This should be part of the default config IMO (the registry bit)

File: modules/local/dada2/main.nf [hunk 3]
```diff
@@ -11,7 +11,7 @@ process DADA2 {
     output:
     tuple val(meta), path("*map.txt"), path("*asvs.fasta"), path("*_filt.fastq.gz"), emit: dada2_out
     path("*asv_counts.tsv")                                                        , emit: dada2_asv_counts
```
Comment (mberacochea) [inline modules/local/dada2/main.nf:13]: Should'nt `asv_counts` also have the `vel(meta)`?

File: modules/nf-core/custom/dumpsoftwareversions/templates/dumpsoftwareversions.py [hunk 4]
```diff
@@ -77,16 +79,15 @@ def main():
...
     versions_by_module["Workflow"] = {
         "Nextflow": "$workflow.nextflow.version",
-        "$workflow.manifest.name": "$workflow.manifest.version",
     }
 
     versions_mqc = {
         "id": "software_versions",
-        "section_name": "${workflow.manifest.name} Software Versions",
-        "section_href": "https://github.com/${workflow.manifest.name}",
+        "section_name": "Pipeline Software Versions",
```
Comment (mberacochea) [inline modules/nf-core/custom/dumpsoftwareversions/templates/dumpsoftwareversions.py:86]: I would revert this change, and add a manifest section to the nextflow.config
```
manifest {
    name            = 'ebi-metagenomics/amplicon-pipeline'
    author          = """Microbiome Informatics Team - EMBL-EBI"""
    homePage        = 'https://github.com/EBI-Metagenomics/amplicon-pipeline'
    description     = """Microbiome Informatics metagenomes amplicon analysis pipeline"""
    mainScript      = 'main.nf'
    nextflowVersion = '!>=23.04.0' ---
    version         = '1.0.0' ---
    doi             = ''
}
```

Comment (mberacochea) [reply modules/nf-core/custom/dumpsoftwareversions/templates/dumpsoftwareversions.py:86]: nextflow loads this when running the pipeline

File: workflows/pipeline.nf [hunk 5]
```diff
@@ -243,4 +246,37 @@ workflow AMPLICON_PIPELINE {
         ch_versions.unique().collectFile(name: 'collated_versions.yml')
     )
 
+    multiqc_input = CONCAT_PRIMER_CUTADAPT.out.cutadapt_json.map{ meta, json ->
```
Comment (mberacochea) [inline workflows/pipeline.nf:249]: Good stuff

**Review by mberacochea — CHANGES_REQUESTED** [review]
Great stuff! I left a few comments.

More generally I think we need to decide on which bits of the nf-core template we should use in all the pipelines and apply it here.
For example, the multiqc bit is well defined in the nf-core template so we should use that convention (which is similar to what we have here).

File: conf/modules.config [hunk 6]
```diff
@@ -12,28 +12,42 @@
...
+                                return null;
+                            }
+                        }
+                        else {
+                            if ( filename.contains(".merged.fastq.gz") ){
+                                return filename;
+                            }
+                            else{
+                                return null;
+                            }
```
Comment (mberacochea) [inline conf/modules.config:38]:
```suggestion
if ((meta.single_end && filename.contains(".fastp.fastq.gz")) ||
    (!meta.single_end && filename.contains(".merged.fastq.gz"))) {
    return filename;
} else {
    return null;
}
```
I think that less nested code is easier to follow.

File: conf/modules.config [hunk 7]
```diff
@@ -12,28 +12,42 @@
...
-            [
-                path: { "${params.outdir}/${meta.id}/qc/" },
-                mode: params.publish_dir_mode,
-                failOnError: true,
-                pattern: ".json" 
+                pattern: "*.json"
             ]
         ]
-    ext.args = '-l 100' // Filter out reads smaller than 100 bases
+    ext.args = '--cut_front_window_size 1 --cut_tail_window_size 1 --cut_front 3 --cut_tail 3 --cut_right --cut_right_window_size 4 --cut_right_mean_quality 15 -l 100' // Filter out reads smaller than 100 bases
```
Comment (mberacochea) [inline conf/modules.config:50]:
```suggestion
    ext.args = [
    '--cut_front_window_size 1',
    '--cut_tail_window_size 1',
    '--cut_front 3',
    '--cut_tail 3',
    '--cut_right',
    '--cut_right_window_size 4',
    '--cut_right_mean_quality 15',
    '-l 100' // Filter out reads smaller than 100 bases
    ].join(' ');
```
Again, this is just to make it easier to spot which options we are using. One flag with its value per line makes it easier to identify.

File: conf/modules.config [hunk 8]
```diff
@@ -205,19 +219,16 @@ process {
...
+    withName: CUTADAPT {
         publishDir = [
             [
                 path: { "${params.outdir}/${meta.id}/primer-identification/" },
                 mode: params.publish_dir_mode,
                 failOnError: true,
-                pattern: "*.fasta"
-            ]
+                pattern: "*.json"
+            ]
```
Comment (mberacochea) [inline conf/modules.config:229]:
```suggestion
            ]
```

File: conf/modules.config [hunk 9]
```diff
@@ -308,6 +319,51 @@ process {
...
+                        }
+                        else {
+                            return filename;
+                        }
+                    }
+                }
+            ]
+        ]
+        ext.prefix = {"${meta.id}_multiqc_report"}
+
```
Comment (mberacochea) [inline conf/modules.config:364]:
```suggestion
```

File: nextflow.config [hunk 10]
```diff
@@ -90,6 +88,22 @@ profiles {
...
+}
+
+manifest {
+    name            = 'ebi-metagenomics/amplicon-pipeline'
+    author          = """Microbiome Informatics Team - EMBL-EBI"""
+    homePage        = 'https://github.com/EBI-Metagenomics/amplicon-pipeline'
+    description     = """Microbiome Informatics metagenomes amplicon analysis pipeline"""
+    mainScript      = 'main.nf'
+    nextflowVersion = '!>=23.04.0'
+    version         = '1.0.0'
```
Comment (mberacochea) [inline nextflow.config:105]: Is this the right version?

**Review by mberacochea — COMMENTED** [review]
Thanks. I left 4 tiny comments other than that it looks good to me.

File: nextflow.config (hunk 10 above)
Comment (chrisAta) [reply nextflow.config:105]: the pipeline has never really had a real version yet... might as well start with 1.0.0 I think?

Comment (mberacochea) [reply nextflow.config:105]: eah, I wasn't sure about the version. 1.0.0 seems like a good choice; it has had plenty of testing up to this point.

**Review by mberacochea — APPROVED** [review]
Thanks Chris, this is really good stuff.
